pkg/worker/workflow.go

```diff
@@ -358,6 +432,36 @@ func (w *worker) ComponentActivity(ctx context.Context, param *ComponentActivity
 	logger, _ := logger.GetZapLogger(ctx)
 	logger.Info("ComponentActivity started")
 
+	// Define activity options
+	ao := workflow.ActivityOptions{
```

Comment:
@donch1989 do you know what this is for?

Reply:
oh I guess it's the configuration for a temporal task.
the ctx below does not match the type needed. do u have any idea how to fix that?

Reply:
This is trying to call a Temporal Activity from another Activity, which goes against how Temporal Workflows are structured. It should be 2 separate activities [called from a Workflow](https://community.temporal.io/t/how-to-call-an-activity-from-other-activity/1294). If one depends on the other's data, the Workflow should orchestrate this.